src/Aspire.Hosting.Sdk/SDK/Sdk.targets

```diff
@@ -13,34 +15,92 @@
+        all</ExcludeAssets>
       <Private Condition="'%(Private)' == '' and '%(IsAspireProjectResource)' == 'true'">false</Private>
     </ProjectReference>
-    
+
   </ItemGroup>
 
-  <Target Name="__WarnOnAspireCapabilityMissing" BeforeTargets="PrepareForBuild" Condition="!@(ProjectCapability->AnyHaveMetadataValue('Identity', 'Aspire'))">
-    <Warning Code="ASPIRE002" Text="$(MSBuildProjectName) is an Aspire AppHost project but necessary dependencies aren't present. Are you missing an Aspire.Hosting.AppHost PackageReference?" />
+  <Target Name="__WarnOnAspireCapabilityMissing" BeforeTargets="PrepareForBuild"
+    Condition="!@(ProjectCapability->AnyHaveMetadataValue('Identity', 'Aspire'))">
+    <Warning Code="ASPIRE002"
+      Text="$(MSBuildProjectName) is an Aspire AppHost project but necessary dependencies aren't present. Are you missing an Aspire.Hosting.AppHost PackageReference?" />
+  </Target>
+
+  <Target Name="__WarnOnMininumVsVersionMissing" BeforeTargets="PrepareForBuild"
+    Condition="'$(BuildingInsideVisualStudio)' == 'true' and $([MSBuild]::VersionLessThan('$(MSBuildVersion)', '17.10.0'))">
+    <Warning Code="ASPIRE003"
+      Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that requires Visual Studio version 17.10 or above to work correctly. You are using version $(MSBuildVersion)." />
   </Target>
 
-  <Target Name="__WarnOnMininumVsVersionMissing" BeforeTargets="PrepareForBuild" Condition="'$(BuildingInsideVisualStudio)' == 'true' and $([MSBuild]::VersionLessThan('$(MSBuildVersion)', '17.10.0'))">
-    <Warning Code="ASPIRE003" Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that requires Visual Studio version 17.10 or above to work correctly. You are using version $(MSBuildVersion)." />
+  <!-- This target extracts the version of Aspire.Hosting.AppHost referenced by the project, and adds
+       a reference to Aspire.Dashboard.Sdk and Aspire.Hosting.Orchestration for the build-time platform using
+       the same version. This is done here dynamically to avoid having to pull in DCP and Dashboard packages
+       for all of the platforms. This mechanism can be disabled by setting `SkipAddReferenceToDashboardAndDCP` to `true` -->
+  <Target Name="AddReferenceToDashboardAndDCP" BeforeTargets="Restore;CollectPackageReferences" Condition="'$(SkipAddReferenceToDashboardAndDCP)' != 'true'">
+
+    <!-- First, we assume project is not using Central Package Management, and we try to extract the version
+         from PackageReference Items. -->
+    <ItemGroup>
+      <AppHostPackageReference Include="@(PackageReference)" Condition="'%(Identity)' ==
+  'Aspire.Hosting.AppHost'" />
+    </ItemGroup>
+
+    <!-- Extract the Version metadata -->
+    <PropertyGroup>
+      <AppHostVersion>%(AppHostPackageReference.Version)</AppHostVersion>
+    </PropertyGroup>
+
+    <!-- If the version is still empty, then we assume the project is using Central Package Management, 
```

Comment:
I don't know CPM enough to see if this will always be the case (as-in not sure if you could have PackageVersion without setting that). If you know that for sure, I can change it, or we can change in a follow-up.

Reply:
That makes sense. I was mainly thinking about the case where we might be able to "know" that when CPM is enabled, and a `PackageVersion` must be available, and thus show a good error.